mittab/apps/tab/tab_card.py

```diff
@@ -0,0 +1,284 @@
+from decimal import Decimal
+import json
+from mittab.apps.tab.helpers import redirect_and_flash_error
+from mittab.apps.tab.models import Bye, RoundStats
+from mittab.apps.tab.models import Team, Round, TabSettings, Debater, Bye, RoundStats
+from django.db.models import Prefetch
+
+from mittab.libs.tab_logic.stats import tot_ranks, tot_ranks_deb, tot_speaks, tot_speaks_deb
+
+def get_victor_label(victor_code, side):
+    side = 0 if side == "G" else 1
+    victor_map = {
+        1: ("W", "L"),
+        2: ("L", "W"),
+        3: ("WF", "LF"),
+        4: ("LF", "WF"),
+        5: ("AD", "AD"),
+        6: ("AW", "AW"),
+    }
+    return victor_map[victor_code][side]
+
+def get_dstats(round_obj, deb1, deb2, iron_man):
+    """Taken from original tab card function.
+    Pretty sure there is a good refactor here but seems hard to test and worried about breaking something"""
+    dstat1 = [
+            k for k in RoundStats.objects.filter(debater=deb1).filter(
+                round=round_obj).all()
+        ]
+    dstat2 = []
+    if not iron_man:
+        dstat2 = [
+            k for k in RoundStats.objects.filter(debater=deb2).filter(
+                round=round_obj).all()
+        ]
+    blank_rs = RoundStats(debater=deb1, round=round_obj, speaks=0, ranks=0)
+    while len(dstat1) + len(dstat2) < 2:
+        # Something is wrong with our data, but we don't want to crash
+        dstat1.append(blank_rs)
+    if not dstat2 and not dstat1:
+        return None, None
+    if not dstat2:
+        dstat1, dstat2 = dstat1[0], dstat1[1]
+    elif not dstat1:
+        dstat1, dstat2 = dstat2[0], dstat2[1]
+    else:
+        dstat1, dstat2 = dstat1[0], dstat2[0]
+    return dstat1, dstat2
+
+def json_get_round(round_obj, team, deb1, deb2):
+    jsonRound = {
+        "round_number": round_obj.round_number,
+        "round_id": round_obj.pk,
+        "side": "G" if round_obj.gov_team == team else "O",
+        "result": get_victor_label(round_obj.victor, "G" if round_obj.gov_team == team else "O"),
+        "chair": round_obj.chair.name,
+        "wings": [judge.name for judge in round_obj.judges.exclude(pk=round_obj.chair.pk)]
```

Comment:
[need to double check this] I _think_ `exclude` will lead to a SQL query since it's a Django ORM method, we're better off filtering in-memory, e.g. `list(filter(round_obj.judges, lambda j: j.pk != round_obj.chair.pk)`